src/tests/ftest/util/server_utils.py

```diff
@@ -48,6 +53,193 @@ class ServerFailed(Exception):
+        names.extend(["request", "action"])
+        return names
+
+    def get_params(self, test, path="/run/daos_server/*"):
+        """Get values for all of the server command params using a yaml file.
+
+        Sets each BasicParameter object's value to the yaml key that matches
+        the assigned name of the BasicParameter object in this class. For
+        example, the self.block_size.value will be set to the value in the yaml
+        file with the key 'block_size'.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace.
+                Defaults to "/run/daos_server/*".
+
+        """
+        super(ServerCommand, self).get_params(test, path)
+
+    def get_launch_command(self, manager, uri=None, env=None):
+        """Get the process launch command used to run daos_server.
+
+        Args:
+            manager (str): mpi job manager command
+            uri (str, optional): path to uri file. Defaults to None.
+            env (dict, optional): dictionary on env variable names and values.
+                Defaults to None.
+
+        Raises:
+            ServerFailed: if the specified job manager is unsupported.
+
+        Returns:
+            str: daos_sever launch command
+
+        """
+        if manager is None:
+            # Run locally
+            return self.__str__()
+
+        elif manager.endswith("orterun"):
+            args = []
+            if env:
+                assign_env = [
+                    "{}={}".format(key, val) for key, val in env.items()]
+                args = [
+                    "-np {}".format(len(self.hosts)),
+                    "-hostfile {}".format(self.hostfile),
+                    "--enable-recovery",
+                ]
+                args.extend(["-x {}".format(assign) for assign in assign_env])
+            if uri is not None:
+                args.append("--report-uri {}".format(uri))
+
+            return "{} {} {}".format(manager, " ".join(args), self.__str__())
+        else:
+            raise ServerFailed("Unsupported job manager: {}".format(manager))
+
+    def start(self, manager, verbose=True, env=None, timeout=600):
+        """Start the daos server on each specified host.
+
+        Args:
+            manager (str): mpi job manager command
+            verbose (bool, optional): [description]. Defaults to True.
+            env ([type], optional): [description]. Defaults to None.
+            timeout (int, optional): [description]. Defaults to 600.
+
+        Raises:
+            ServerFailed: if there are issues starting the servers
+
+        """
+        if self.process is None:
+            # Start the daos server as a subprocess
+            kwargs = {
+                "cmd": self.get_launch_command(manager),
+                "verbose": verbose,
+                "allow_output_check": "combined",
+                "shell": True,
+                "env": env,
+                "sudo": True,
```

Comment:
Why are we using sudo here?  The existing behaviour is to run the servers as non-root, isn't it?

Reply:
It is.  Having it run as sudo for this test was an oversight of mine...  I'll create an argument for sudo as I will need that functionality for future tests to verify that the server behaves correctly when starting both as non-root and as root. For this test, sudo is not needed.